core/src/main/java/org/elasticsearch/index/query/TermsQueryBuilder.java

```diff
@@ -32,106 +58,121 @@
 
     static final TermsQueryBuilder PROTOTYPE = new TermsQueryBuilder(null, (Object) null);
 
-    private final String name;
+    public static final boolean DEFAULT_DISABLE_COORD = false;
 
-    private final Object values;
-
-    private String execution;
+    private final String fieldName;
+    private List<Object> values;
+    private boolean disableCoord;
 
     private String lookupIndex;
     private String lookupType;
     private String lookupId;
-    private String lookupRouting;
     private String lookupPath;
```

Comment:
how about introducing a writeable object that holds the lookup info? then to determine if it's a lookup or not would be checking if that object is null or not. we could have a new setter for that new object, and also keep the existing setters that create a new instance if null (for bw comp).